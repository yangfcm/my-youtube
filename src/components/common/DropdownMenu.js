import React from "react";
import { Link } from "react-router-dom";
import GoogleAuth from "../modules/GoogleAuth";

const DropdownMenu = ({ menuItems = [] }) => {
  return (
    <div className="list-group">
      {menuItems.map((item) => {
        return (
          <Link
            key={item.name}
            to={item.link}
            className="list-group-item list-group-item-action"
          >
            {item.name}
          </Link>
        );
      })}
      <GoogleAuth />
    </div>
  );
};

export default DropdownMenu;
