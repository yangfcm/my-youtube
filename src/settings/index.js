import axios from "axios";

export const apiBaseUrl = "https://www.googleapis.com/youtube/v3";

export const defAxios = axios.create({
  baseURL: apiBaseUrl,
  timeout: 10000,
});

export const defaultName = "Y Fan";
export const mainMenuItems = [
  { link: "/", exact: true, name: "Recommend", icon: "thumbs-up" },
  { link: "/channel", exact: false, name: "Channel", icon: "user-plus" },
  { link: "/playlist", exact: false, name: "Play List", icon: "list" },
];

export const maxResults = 12;
