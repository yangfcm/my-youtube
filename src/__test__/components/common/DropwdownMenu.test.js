import React from "react";
import { Provider } from "react-redux";
import { Router, Link } from "react-router-dom";
import { createBrowserHistory } from "history";
import { mount } from "enzyme";
import configMockStore from "redux-mock-store";
import DropdownMenu from "components/common/DropdownMenu";
import { mainMenuItems } from "settings";

describe("Test DropdownMenu component", () => {
  let wrapper, mockStore;
  const googleSignoutSpy = jest.fn();
  const history = createBrowserHistory();

  beforeEach(() => {
    mockStore = configMockStore([]);
    wrapper = mount(
      <Provider store={mockStore({})}>
        <Router history={history}>
          <DropdownMenu
            googleSignout={googleSignoutSpy}
            menuItems={mainMenuItems}
          />{" "}
        </Router>
      </Provider>
    );
  });

  it("should render DropdownMenu component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render three links", () => {
    expect(wrapper.find(Link)).toHaveLength(mainMenuItems.length);
  });

  it("should have Sign out link", () => {
    expect(wrapper.find("li.list-group-item-warning").exists()).toBe(true);
  });

  it("click Sign out link should call googleSignout function", () => {
    wrapper.find("li.list-group-item-warning").simulate("click");
    expect(googleSignoutSpy).toHaveBeenCalled();
  });
});
