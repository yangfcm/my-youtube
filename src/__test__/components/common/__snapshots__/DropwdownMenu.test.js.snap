// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DropdownMenu component should render DropdownMenu component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(GoogleAuthWrapper)
      googleSignout={[MockFunction]}
      menuItems={
        Array [
          Object {
            "exact": true,
            "icon": "thumbs-up",
            "link": "/",
            "name": "Recommend",
          },
          Object {
            "exact": false,
            "icon": "user-plus",
            "link": "/channel",
            "name": "Subscriptions",
          },
          Object {
            "exact": false,
            "icon": "list",
            "link": "/playlist",
            "name": "Play List",
          },
        ]
      }
    >
      <GoogleAuthWrapper
        googleSignout={[MockFunction]}
        menuItems={
          Array [
            Object {
              "exact": true,
              "icon": "thumbs-up",
              "link": "/",
              "name": "Recommend",
            },
            Object {
              "exact": false,
              "icon": "user-plus",
              "link": "/channel",
              "name": "Subscriptions",
            },
            Object {
              "exact": false,
              "icon": "list",
              "link": "/playlist",
              "name": "Play List",
            },
          ]
        }
        signIn={[Function]}
        signOut={[Function]}
      >
        <DropdownMenu
          googleSignin={[Function]}
          googleSignout={[MockFunction]}
          googleUser={null}
          menuItems={
            Array [
              Object {
                "exact": true,
                "icon": "thumbs-up",
                "link": "/",
                "name": "Recommend",
              },
              Object {
                "exact": false,
                "icon": "user-plus",
                "link": "/channel",
                "name": "Subscriptions",
              },
              Object {
                "exact": false,
                "icon": "list",
                "link": "/playlist",
                "name": "Play List",
              },
            ]
          }
          signIn={[Function]}
          signOut={[Function]}
        >
          <div
            className="list-group"
          >
            <Link
              className="list-group-item list-group-item-action"
              key="Recommend"
              to="/"
            >
              <LinkAnchor
                className="list-group-item list-group-item-action"
                href="/"
                navigate={[Function]}
              >
                <a
                  className="list-group-item list-group-item-action"
                  href="/"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="thumbs-up"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-thumbs-up fa-w-16 "
                      data-icon="thumbs-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                     
                  Recommend
                </a>
              </LinkAnchor>
            </Link>
            <Link
              className="list-group-item list-group-item-action"
              key="Subscriptions"
              to="/channel"
            >
              <LinkAnchor
                className="list-group-item list-group-item-action"
                href="/channel"
                navigate={[Function]}
              >
                <a
                  className="list-group-item list-group-item-action"
                  href="/channel"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="user-plus"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-user-plus fa-w-20 "
                      data-icon="user-plus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 640 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                     
                  Subscriptions
                </a>
              </LinkAnchor>
            </Link>
            <Link
              className="list-group-item list-group-item-action"
              key="Play List"
              to="/playlist"
            >
              <LinkAnchor
                className="list-group-item list-group-item-action"
                href="/playlist"
                navigate={[Function]}
              >
                <a
                  className="list-group-item list-group-item-action"
                  href="/playlist"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="list"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-list fa-w-16 "
                      data-icon="list"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                     
                  Play List
                </a>
              </LinkAnchor>
            </Link>
            <li
              className="list-group-item list-group-item-warning"
              onClick={[MockFunction]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="sign-out-alt"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-sign-out-alt fa-w-16 "
                  data-icon="sign-out-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
                 
              Sign out
            </li>
          </div>
        </DropdownMenu>
      </GoogleAuthWrapper>
    </Connect(GoogleAuthWrapper)>
     
  </Router>
</Provider>
`;
